test/hooks/useSlidingSyncRoomSearch-test.tsx

```diff
@@ -72,40 +60,34 @@ describe("useSlidingSyncRoomSearch", () => {
             }
         });
 
-        // first check that everything is empty and then do the search
-        let executeHook = (roomSearch: RoomSearchHook) => {
-            expect(roomSearch.loading).toBe(false);
-            expect(roomSearch.rooms).toEqual([]);
-            roomSearch.search({
-                limit: 10,
-                query: "foo",
-            });
+        // first check that everything is empty
+        const { result } = renderHook(() => useSlidingSyncRoomSearch());
+        const query = {
+            limit: 10,
+            query: "foo",
         };
-        const wrapper = mount(
-            <RoomSearchComponent
-                onClick={(roomSearch: RoomSearchHook) => {
-                    executeHook(roomSearch);
-                }}
-            />,
-        );
+        expect(result.current.loading).toBe(false);
+        expect(result.current.rooms).toEqual([]);
 
         // run the query
-        await act(async () => {
-            await sleep(1);
-            wrapper.simulate("click");
-            return act(() => sleep(1));
+        act(() => {
+            result.current.search(query);
         });
+
+        // wait for loading to finish
+        await waitFor(() => {
+            expect(result.current.loading).toBe(false);
+        });
+
         // now we expect there to be rooms
-        executeHook = (roomSearch) => {
-            expect(roomSearch.loading).toBe(false);
-            expect(roomSearch.rooms).toEqual([roomA, roomB]);
-        };
+        expect(result.current.rooms).toEqual([roomA, roomB]);
 
-        // run the query
-        await act(async () => {
-            await sleep(1);
-            wrapper.simulate("click");
-            return act(() => sleep(1));
+        // run the query again
+        act(() => {
+            result.current.search(query);
+        });
+        await waitFor(() => {
+            expect(result.current.loading).toBe(false);
```

Comment:
I can't for the life of me figure out why we rerun it, but have left it in to show it's converting like for like